DependencyInjection/BernardExtension.php

```diff
@@ -15,40 +15,50 @@ protected function loadInternal(array $config, ContainerBuilder $container)
         $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
         $loader->load('services.xml');
 
-        $container->setAlias('bernard.driver', 'bernard.driver.'.$config['driver']);
-
         switch ($config['driver']) {
             case 'doctrine':
+                $container->setAlias('bernard.driver', 'bernard.driver.'.$config['driver']);
                 $this->registerDoctrineConfiguration($config['options'], $container);
                 break;
 
             case 'file':
+                $container->setAlias('bernard.driver', 'bernard.driver.'.$config['driver']);
                 $this->registerFlatFileConfiguration($config['options'], $container);
                 break;
 
             case 'phpamqp':
+                $container->setAlias('bernard.driver', 'bernard.driver.'.$config['driver']);
                 $this->registerPhpAmqpConfiguration($config['options'], $container);
                 break;
 
             case 'phpredis':
+                $container->setAlias('bernard.driver', 'bernard.driver.'.$config['driver']);
                 $this->registerPhpRedisConfiguration($config['options'], $container);
                 break;
 
             case 'predis':
+                $container->setAlias('bernard.driver', 'bernard.driver.'.$config['driver']);
                 $this->registerPredisConfiguration($config['options'], $container);
                 break;
 
             case 'ironmq':
+                $container->setAlias('bernard.driver', 'bernard.driver.'.$config['driver']);
                 $this->registerIronMQConfiguration($config['options'], $container);
                 break;
 
             case 'sqs':
+                $container->setAlias('bernard.driver', 'bernard.driver.'.$config['driver']);
                 $this->registerSqsConfiguration($config['options'], $container);
                 break;
 
             case 'pheanstalk':
+                $container->setAlias('bernard.driver', 'bernard.driver.'.$config['driver']);
```

Comment:
it is a bit excessive to have this in all of the cases. Can we centralize it after the switch?